fastfield_codecs/src/serialize.rs

```diff
@@ -167,7 +190,11 @@ pub fn serialize_u128<F: Fn() -> I, I: Iterator<Item = u128>>(
     num_vals: u32,
     output: &mut impl io::Write,
 ) -> io::Result<()> {
-    // TODO write header, to later support more codecs
+    let header = U128Header {
+        num_vals,
+        codec_type: U128FastFieldCodecType::CompactSpace,
+    };
+    header.serialize(output)?;
```

Comment:
Can you add a dumb unit test testing `deserialize(serialize(T)) == T` on a simple sample?